### PR TITLE
pygmt.select: Add aliases for "C", "F", and "L"

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PR aims to add aliases for `pygmt.select`:
- [ ] [**C**](https://docs.generic-mapping-tools.org/dev/gmtselect.html#c) -> ~`pointfile`~ `dist2pt` 
- [ ] [**F**](https://docs.generic-mapping-tools.org/dev/gmtselect.html#f) -> ~`polygonfile`~ `polygon`
- [ ] [**L**](https://docs.generic-mapping-tools.org/dev/gmtselect.html#l) -> ~`line_file`~  `dist2line`

The aliases are currently based on the Julia documentation at
https://joa-quim.github.io/GMTjl_doc/documentation/modules/gmtselect/index.html.

**Preview**: https://pygmt-dev--2466.org.readthedocs.build/en/2466/api/generated/pygmt.select.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/src/select.py [hunk 1]
```diff
@@ -128,6 +131,28 @@ def select(data=None, outfile=None, **kwargs):
...
+        Pass all records whose locations are within *dist* of any of the
+        points in the ASCII file *pointfile*. If *dist* is zero, the 3rd
+        column of *pointfile* must have each point’s individual radius of
+        influence. If you only have a single point you can specify
+        *lon*/*lat* instead of *pointfile*. Distances are Cartesian and in
+        user units.
+    dist2line : str
+        *linefile*\ **+d**\ *dist*\ [**+p**].
+        Pass all records whose locations are within *dist* of any of the
+        line segments in the ASCII multiple-segment file *linefile*.
```
Comment (yvonnefroehlich) [inline pygmt/src/select.py]: I am wondering, whether all users (especially non GMT users) are familiar with the term "multiple-segment file"?

File: pygmt/src/select.py [hunk 2]
```diff
@@ -128,6 +131,38 @@ def select(data=None, outfile=None, **kwargs):
...
         given range. Finally, if **+c** is not used then it is automatically
         incremented for each new ``z_subregion`` argument, starting with 2.
+    dist2pt : str
+        *pointfile*\|\ *lon*/*lat*\ **+d**\ *dist*.
+        Pass all records whose locations are within *dist* of any of the
+        points in the ASCII file *pointfile*. If *dist* is zero, the 3rd
+        column of *pointfile* must have each point's individual radius of
+        influence. If you only have a single point you can specify
+        *lon*/*lat* instead of *pointfile*. Distances are Cartesian and in
+        user units. Alternatively, if ``region`` and ``projection`` are used,
```
Comment (yvonnefroehlich) [inline pygmt/src/select.py:141]: I am unsure about the **-fg** option in the [upstream GMT documenation](https://docs.generic-mapping-tools.org/dev/gmtselect.html#c):
> Distances are Cartesian and in user units; specify -fg to indicate spherical distances and append a distance unit, even if the distance specified is 0.

File: pygmt/src/select.py [hunk 3]
```diff
@@ -128,6 +131,38 @@ def select(data=None, outfile=None, **kwargs):
...
+        centimeters, inches, meters, or points, as determined by
+        :gmt-term:`PROJ_LENGTH_UNIT`) before Cartesian distances are compared
+        to *dist*.
+    dist2line : str
+        *linefile*\ **+d**\ *dist*\ [**+p**].
+        Pass all records whose locations are within *dist* of any of the line
+        segments in the ASCII multiple-segment file *linefile*. If *dist* is
+        zero, we will scan each sub-header in *linefile* for an embedded
+        **-D**\ *dist* setting that sets each line's individual distance value.
+        Distances are Cartesian and in user units. Alternatively, if ``region``
```
Comment (yvonnefroehlich) [inline pygmt/src/select.py:152]: I am unsure about the **-fg** option in the [upstream GMT documenation](https://docs.generic-mapping-tools.org/dev/gmtselect.html#l):
> Distances are Cartesian and in user units; specify -fg to indicate spherical distances append a distance unit (see [Units](https://docs.generic-mapping-tools.org/dev/gmtselect.html#units)).

**Review by michaelgrund — APPROVED** [review]

File: pygmt/src/select.py (hunk 1 above)
Comment (weiji14) [reply pygmt/src/select.py]: Could link to https://docs.generic-mapping-tools.org/6.4/cookbook/file-formats.html#optional-segment-header-records perhaps?

File: pygmt/src/select.py [hunk 4]
```diff
@@ -128,6 +131,38 @@ def select(data=None, outfile=None, **kwargs):
         and **+i** reverses the tests to pass record with *z* value NOT in the
         given range. Finally, if **+c** is not used then it is automatically
         incremented for each new ``z_subregion`` argument, starting with 2.
+    dist2pt : str
```
Comment (weiji14) [inline pygmt/src/select.py:134]: Could you move the docstring for dist2pt/dist2line/polygon up a bit? Maybe under `area_thresh`?

File: pygmt/src/select.py (hunk 3 above)
Comment (weiji14) [reply pygmt/src/select.py:152]: It sounds like you could set arc degree (d), arc minute (m), arc second (s) as units, but these geographical units will be projected to Cartesian units (e.g. meter) when doing the `select` operation?

File: pygmt/src/select.py (hunk 4 above)
Comment (yvonnefroehlich) [reply pygmt/src/select.py]: This is done in commit https://github.com/GenericMappingTools/pygmt/pull/2466/commits/ba6c65314c797dab431191b0686919ed7b0a0edd.

File: pygmt/src/select.py (hunk 1 above)
Comment (yvonnefroehlich) [reply pygmt/src/select.py]: I added this link in commit https://github.com/GenericMappingTools/pygmt/pull/2466/commits/87138eab42f51b02126b52e7731d7774db841128.

**Review by weiji14 — APPROVED** [review]
Thanks @yvonnefroehlich!

File: pygmt/src/select.py (hunk 3 above)
Comment (yvonnefroehlich) [reply pygmt/src/select.py]: I can try to look at this later this day, but not sure if I find the time 🙁.

Comment (yvonnefroehlich) [reply pygmt/src/select.py]: It is very likely that I will not have the time to look at this in more detail today. So, @weiji14 you can merge this PR to have it within the v0.10.0 release, as it has two approvals already.

Comment (yvonnefroehlich) [reply pygmt/src/select.py]: Sounds good.

**Pull request merged**